### PR TITLE
Updates guac template to include cloudwatch agent to collect logs

### PR DESCRIPTION
**Pull request by allez-allez-allez**

### CODE REVIEW

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml [hunk 1]
```diff
@@ -89,6 +93,18 @@ Parameters:
     Default: ''
     Description: Text/Label to display branding for the Guac Login page
     Type: String
+  CloudWatchAgentUrl:
+    AllowedPattern: '^$|^s3://.*\.rpm$'
+    Default: ''
+    Description: >-
+      (Optional) S3 URL to CloudWatch Agent installer. Example(CentOS):
```
Comment (lorengordon) [inline templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: The rpm is common to any EL-derivative, not just CentOS, and this whole template is dependent on being an EL-derivative...

```suggestion
      (Optional) S3 URL to CloudWatch Agent installer. Example:
```

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml [hunk 2]
```diff
@@ -89,6 +93,18 @@ Parameters:
...
+      s3://amazoncloudwatch-agent/amazon_linux/amd64/latest/amazon-cloudwatch-agent.rpm
+    Type: String
+  CloudWatchAppLogs:
```
Comment (lorengordon) [inline templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: This template is _specific_ to the guac app, so we don't really need to expose a parameter for the logs we want the app to send, we can just include them in the template.

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml [hunk 3]
```diff
@@ -432,12 +450,108 @@ Resources:
...
             - setup
             - make-guac
             - finalize
+        cw-agent-install:
+          commands:
+            01-install-cloudwatch-agent:
+              command: !Sub >-
+                aws s3 cp ${CloudWatchAgentUrl} /etc/cfn/scripts/amazon-cloudwatch-agent.rpm &&
+                rpm -v -U /etc/cfn/scripts/amazon-cloudwatch-agent.rpm
+            03-manipulate-cloudwatch-config:
```
Comment (lorengordon) [inline templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: We can remove this, per comment on the parameter

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml [hunk 4]
```diff
@@ -432,12 +450,108 @@ Resources:
...
           update:
             - setup
             - make-guac
             - finalize
+        cw-agent-install:
+          commands:
+            01-install-cloudwatch-agent:
+              command: !Sub >-
+                aws s3 cp ${CloudWatchAgentUrl} /etc/cfn/scripts/amazon-cloudwatch-agent.rpm &&
+                rpm -v -U /etc/cfn/scripts/amazon-cloudwatch-agent.rpm
```
Comment (lorengordon) [inline templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: not sure why we went with rpm elsewhere. try:

```suggestion
                yum localinstall /etc/cfn/scripts/amazon-cloudwatch-agent.rpm
```

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml [hunk 5]
```diff
@@ -432,12 +450,108 @@ Resources:
...
+                  {
+                    "logs_collected":
+                    {
+                      "files":
+                      {
+                        "collect_list": [
+                          {
+                            "file_path": "/opt/aws/amazon-cloudwatch-agent/logs/amazon-cloudwatch-agent.log",
+                            "log_group_name": "/aws/ec2/lx/${AWS::StackName}",
+                            "log_stream_name": "cloudwatch_agent_logs_{instance_id}",
```
Comment (lorengordon) [inline templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: _Just_ updated this log stream name in the wam templates, can you follow the new pattern for log stream names?  `{instance_id}//<path>...` See: https://github.com/plus3it/terraform-aws-watchmaker/pull/144

File: templates/ra_guac_autoscale_public_alb.template.cfn.yaml [hunk 6]
```diff
@@ -432,12 +450,108 @@ Resources:
...
+                            "log_stream_name": "messages_logs_{instance_id}",
+                            "timestamp_format": "%H:%M:%S %y %b %-d"
+                          }
+                        ]
+                      }
+                    },
+                    "log_stream_name": "default_logs_{instance_id}"
+                  }
+                }
+            /etc/cfn/scripts/aws-cw-agent/cloudwatch-applog-config.py:
```
Comment (lorengordon) [inline templates/ra_guac_autoscale_public_alb.template.cfn.yaml]: Can remove, since we don't need to merge parameter-specified log file paths

**Comment by lorengordon** [comment]
@allez-allez-allez before merging, can you please also `bumpversion patch`?

**Comment by allez-allez-allez** [comment]
version bumped. ready for merge

**Pull request merged**